packages/experimental/Menu/src/MenuItem/useMenuItem.ts

```diff
@@ -1,34 +1,71 @@
 import * as React from 'react';
-import { AccessibilityState } from 'react-native';
+import { AccessibilityState, I18nManager } from 'react-native';
 import { MenuItemProps, MenuItemState } from './MenuItem.types';
 import { memoize } from '@fluentui-react-native/framework';
-import { useAsPressable, useKeyProps } from '@fluentui-react-native/interactive-hooks';
+import { InteractionEvent, isKeyPressEvent, useAsPressable, useKeyDownProps } from '@fluentui-react-native/interactive-hooks';
 import { useMenuContext } from '../context/menuContext';
 import { useMenuListContext } from '../context/menuListContext';
 import { useMenuTriggerContext } from '../context/menuTriggerContext';
 
+const triggerKeys = [' ', 'Enter'];
+const submenuTriggerKeys = [...triggerKeys, 'ArrowLeft', 'ArrowRight'];
+
 export const useMenuItem = (props: MenuItemProps): MenuItemState => {
   // attach the pressable state handlers
   const defaultComponentRef = React.useRef(null);
   const { onClick, accessibilityState, componentRef = defaultComponentRef, disabled, ...rest } = props;
-  const pressable = useAsPressable({ ...rest, disabled, onPress: onClick });
-  const onKeyProps = useKeyProps(onClick, ' ', 'Enter');
   const isTrigger = useMenuTriggerContext();
-  const hasSubmenu = useMenuContext().isSubmenu && isTrigger;
+  const isSubmenu = useMenuContext().isSubmenu;
+  const hasSubmenu = isSubmenu && isTrigger;
+  const isInSubmenu = isSubmenu && !isTrigger;
+
+  const setOpen = useMenuContext().setOpen;
+  const onInvoke = React.useCallback(
+    (e: InteractionEvent) => {
+      if (!disabled) {
+        if (
+          isKeyPressEvent(e) &&
+          hasSubmenu &&
+          ((I18nManager.isRTL && e.nativeEvent.key === 'ArrowRight') || (!I18nManager.isRTL && e.nativeEvent.key === 'ArrowLeft'))
+        ) {
+          return;
+        }
+        if (
+          isKeyPressEvent(e) &&
+          isInSubmenu &&
+          ((I18nManager.isRTL && e.nativeEvent.key === 'ArrowLeft') || (!I18nManager.isRTL && e.nativeEvent.key === 'ArrowRight'))
+        ) {
+          return;
+        }
```

Comment:
If we have just `return` in both conditions maybe it's better to put it to one? And long conditions with RTL put to variables? I'm just wondering. Not sure if it's better